src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpClientIntegration.cs

```diff
@@ -0,0 +1,772 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.Emit;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.Integrations
+{
+    /// <summary>
+    /// Tracer integration for HttpClient.
+    /// </summary>
+    public static class HttpClientIntegration
+    {
+        private const string IntegrationName = "HttpClient";
+        private const string SystemNetHttp = "System.Net.Http";
+        private const string Major4 = "4";
+        private const string HttpClientTarget = "System.Net.Http.HttpClient";
+        private const string DeleteAsync = "DeleteAsync";
+        private const string SendAsync = "SendAsync";
+        private const string GetAsync = "GetAsync";
+        private const string GetByteArrayAsync = "GetByteArrayAsync";
+        private const string GetStreamAsync = "GetStreamAsync";
+        private const string GetStringAsync = "GetStringAsync";
+        private const string PostAsync = "PostAsync";
+        private const string PutAsync = "PutAsync";
```

Comment:
I think we should remove all of these strings and instead match the wrapper method name to the target method name. In the old `HttpMessageHandlerIntegration` integration we are removing, we had to use `HttpMessageHandler_SendAsync()` and `HttpClientHandler_SendAsync()` because the target method names were the same for both classes (`SendAsync()`). Because the wrapper method name didn't match the target method name, we had to specify the `TargetMethod = "SendAsync"` in the the `[InterceptMethod]` attribute.

In this new integration, we are only instrumenting type `HttpClient`, and each wrapper method has a different name, so we can now match the target's name. For example, for `GetStringAsync()`, we can use:
```csharp
 [InterceptMethod(
        TargetAssembly = SystemNetHttp,
        TargetType = HttpClientTarget,
        /*  TargetMethod = GetStringAsync, */  /* not needed because method name matches */
        TargetSignatureTypes = new[] { ClrNames.StringTask, ClrNames.Uri },
        TargetMinimumVersion = Major4,
        TargetMaximumVersion = Major4)]
        public static object GetStringAsync(
        ...
```

Reply:
Yes, I agree.